driver_test.go

```diff
@@ -12,26 +12,250 @@
+
+	// open client
+	var db = "projects/"+project+"/instances/"+instance+"/databases/"+dbname;
+	ctx := context.Background()
+	client, err := spanner.NewClient(ctx, db)
+	if err != nil {
+			log.Fatal(err)
+	}
+	defer client.Close()
+
+	// Put strings into spanner.Statement structure
+	var states []spanner.Statement
+	for _,line := range dml {
+		states = append(states, spanner.NewStatement(line))
+	}
+
+	// execute statements
+	_, err = client.ReadWriteTransaction(ctx, func(ctx context.Context, txn *spanner.ReadWriteTransaction) error {
+		stmts := states
+		rowCounts, err := txn.BatchUpdate(ctx, stmts)
+		if err != nil {
+				return err
+		}
+		fmt.Printf("Executed %d SQL statements using Batch DML.\n", len(rowCounts))
+		return nil
+		})
+	if (err != nil) { log.Fatal(err) }
+}
+
+// end client lib funs 
+// ******************* //
+
+// helper funs for tests //
+func mustExecContext(t * testing.T, ctx context.Context, db *sql.DB, query string){
+	_,err := db.ExecContext(ctx, query)
+	if err != nil {
+		debug.PrintStack()
+		t.Fatalf(err.Error())
+	}
+}
+
+func mustQueryContext( t *testing.T, ctx context.Context, db *sql.DB, query string) (rows *sql.Rows){
+	rows, err := db.QueryContext(ctx, query)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	return rows
+}
+
+// Tests general query functionality 
+func TestQueryGeneral(t *testing.T){
+
+	// set up test table
+	curs, err := NewConnector()
+	if err != nil{
+		log.Fatal(err)
+	}
+
+	executeDdlApi(curs, []string{`CREATE TABLE Testa (
+		A   STRING(1024),
+		B  STRING(1024),
+		C   STRING(1024)
+	)	 PRIMARY KEY (A)`}) 
+	ExecuteDMLClientLib([]string{`INSERT INTO Testa (A, B, C) 
+		VALUES ("a1", "b1", "c1"), ("a2", "b2", "c2") , ("a3", "b3", "c3") `}) 
+
+
+	// cases
+	type test struct {
+        input string
+        want  []testaRow
+    }
+
+	tests := []test{
+		// return one row
+		{input: "SELECT * FROM Testa WHERE A = \"a1\"", 
+		want: []testaRow{
+			{A:"a1", B:"b1", C:"c1"},
+		}},
+		// return whole table 
+		{input: "SELECT * FROM Testa ORDER BY A", 
+		want: []testaRow{
+			{A:"a1", B:"b1", C:"c1"},
+			{A:"a2", B:"b2", C:"c2"},
+			{A:"a3", B:"b3", C:"c3"},
+		}},
+	}
+
+	// run tests
+	for _, tc := range tests {
+		got := RunQueryGeneral(t, tc.input)
+        if !reflect.DeepEqual(tc.want, got) {
+            t.Errorf("expected: %v, got: %v", tc.want, got)
+		}
+	}
+
+	// drop table 
+	executeDdlApi(curs, []string{`DROP TABLE Testa`});
+
+	// close connection 
+	curs.Close()
+}
+
+// runs query on Testa table, returns result in testaRow array 
+func RunQueryGeneral(t *testing.T, query string,)([]testaRow){
+
+	ctx := context.Background()
+	db, err := sql.Open("spanner", dsn)
+	if err != nil {
+		debug.PrintStack()
+		log.Fatal(err)
+	}
+
+	rows := mustQueryContext(t, ctx, db, query)
+
+	got := []testaRow{}
+	for rows.Next(){
+		curr := testaRow{A:"", B:"", C:""}
+		if err := rows.Scan(&curr.A, &curr.B, &curr.C); err != nil {
+			t.Error(err.Error())
```

Comment:
You can just pass `err` directly.
```suggestion
			t.Error(err)
```

Reply:
👍